### PR TITLE
feat: add support for gpt4o

### PR DESCRIPTION
**Pull request by matthiasthomas**
📜 Lore

Adding support to gpt4o which came out yesterday May, 13th 2024.

🗝️ Key Changes

🛡️ Main Quests

[x] Add support for gpt4o

### CODE REVIEW

**Comment by matthiasthomas** [comment]
@hupe1980 - Would you mind approving this one 🙏 It would unblock me

File: tiktoken.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ const (
 var ModelPrefixToEncoding = map[string]string{
 	// chat
 	"gpt-4-":         CL100kBase, // e.g., gpt-4-0314, etc., plus gpt-4-32k
+	"gpt-4o-":        CL100kBase, // e.g., gpt-4o-2024-05-13, etc.
```
Comment (hupe1980) [inline tiktoken.go]: Shouldn't it actually be o200k?

File: tiktoken.go [hunk 2]
```diff
@@ -28,6 +29,7 @@ var ModelPrefixToEncoding = map[string]string{
 var ModelToEncoding = map[string]string{
 	// chat
 	"gpt-4":         CL100kBase,
+	"gpt-4o":        CL100kBase,
```
Comment (hupe1980) [inline tiktoken.go]: Same here. o200k?

**Review by hupe1980 — CHANGES_REQUESTED** [review]

File: tiktoken.go (hunk 1 above)
Comment (matthiasthomas) [reply tiktoken.go]: Totally @hupe1980 , sorry about that, didn't realize they changed the tokenizer. It's fixed, I've updated according to the latest from https://github.com/openai/tiktoken/tree/main

**Review by hupe1980 — APPROVED** [review]
LGTM! Thx

**Pull request merged**